servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultHttpLoadBalancerFactory.java

```diff
@@ -0,0 +1,204 @@
+import javax.annotation.Nullable;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Default implementation of {@link HttpLoadBalancerFactory}.
+ *
+ * @param <ResolvedAddress> The type of address after resolution.
+ */
+public final class DefaultHttpLoadBalancerFactory<ResolvedAddress>
+        implements HttpLoadBalancerFactory<ResolvedAddress> {
+    private final LoadBalancerFactory<ResolvedAddress, FilterableStreamingHttpLoadBalancedConnection> rawFactory;
+    private final HttpExecutionStrategy strategy;
+
+    DefaultHttpLoadBalancerFactory(
+            final LoadBalancerFactory<ResolvedAddress, FilterableStreamingHttpLoadBalancedConnection> rawFactory,
+            final HttpExecutionStrategy strategy) {
+        this.rawFactory = rawFactory;
+        this.strategy = strategy;
+    }
+
+    @SuppressWarnings("deprecation")
+    @Override
+    public <T extends FilterableStreamingHttpLoadBalancedConnection> LoadBalancer<T> newLoadBalancer(
+            final String targetResource,
+            final Publisher<? extends Collection<? extends ServiceDiscovererEvent<ResolvedAddress>>> eventPublisher,
+            final ConnectionFactory<ResolvedAddress, T> connectionFactory) {
+        return rawFactory.newLoadBalancer(targetResource, eventPublisher, connectionFactory);
+    }
+
+    @Override
+    public LoadBalancer<FilterableStreamingHttpLoadBalancedConnection> newLoadBalancer(
+            final Publisher<? extends Collection<? extends ServiceDiscovererEvent<ResolvedAddress>>> eventPublisher,
+            final ConnectionFactory<ResolvedAddress, FilterableStreamingHttpLoadBalancedConnection> connectionFactory,
+            final String targetResource) {
+        return rawFactory.newLoadBalancer(eventPublisher, connectionFactory, targetResource);
+    }
+
+    @Override   // FIXME: 0.43 - remove deprecated method
+    public FilterableStreamingHttpLoadBalancedConnection toLoadBalancedConnection(
+            final FilterableStreamingHttpConnection connection) {
+        return new DefaultFilterableStreamingHttpLoadBalancedConnection(connection);
+    }
+
+    @Override
+    public FilterableStreamingHttpLoadBalancedConnection toLoadBalancedConnection(
+            final FilterableStreamingHttpConnection connection,
+            final ReservableRequestConcurrencyController concurrencyController,
+            @Nullable final ContextMap context) {
+        return new HttpLoadBalancerFactory.DefaultFilterableStreamingHttpLoadBalancedConnection(connection,
+                concurrencyController);
+    }
+
+    @Override
+    public HttpExecutionStrategy requiredOffloads() {
+        return strategy;
+    }
+
+    /**
+     * A builder for creating instances of {@link DefaultHttpLoadBalancerFactory}.
+     *
+     * @param <ResolvedAddress> The type of address after resolution for the {@link HttpLoadBalancerFactory} built by
+     * this builder.
+     */
+    public static final class Builder<ResolvedAddress> {
```

Comment:
While we're here, do we really need this builder? The only method on  there is the `build()` and it returns a factory.

Reply:
Good question. Taking into account that we didn't have a need for any other builder options for years, I'm fine with dropping the builder and making the ctor public:
```java
    /**
     * Creates a new instance.
     * 
     * @param rawFactory {@link LoadBalancerFactory} to use
     */
    public DefaultHttpLoadBalancerFactory(
            final LoadBalancerFactory<ResolvedAddress, FilterableStreamingHttpLoadBalancedConnection> rawFactory) {
        this.rawFactory = requireNonNull(rawFactory);
        this.strategy = HttpExecutionStrategy.from(rawFactory.requiredOffloads());
    }
```